From 5895838e53a4d2fcf219e662aa944e3fade97edc Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Sun, 21 Jul 2019 07:53:42 +0200
Subject: [PATCH 350/386] arm64: dts: marvell: enable TDM interfaces numbering

Introduce new DT property for the TDM controller, which
reflects its ID. Update binding documentation accordingly.
It will be used for multiple-devices support.

Change-Id: I1d1a50801ef53928098cbdd1113dfac0590c26ce
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12958
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Yan Markman <ymarkman@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 Documentation/devicetree/bindings/telephony/mvebu-phone.txt | 2 ++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi               | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/Documentation/devicetree/bindings/telephony/mvebu-phone.txt b/Documentation/devicetree/bindings/telephony/mvebu-phone.txt
index c5008e2bf906..ee8043355eb4 100644
--- a/Documentation/devicetree/bindings/telephony/mvebu-phone.txt
+++ b/Documentation/devicetree/bindings/telephony/mvebu-phone.txt
@@ -16,6 +16,8 @@ Required properties:
 
 - interrupts - TDM controller interrupt
 
+- cell-index: TDM controller index
+
 - clocks: phandle for internal gate clock
 
 - clock-names: name associated to the clocks: "gateclk"
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index b9a37b6e9eb4..6b7a05a665a9 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -17,6 +17,7 @@
 					 CP110_PCIE_MEM_SIZE(iface))
 
 #define CP110_EIP197_INDEX		CP110_NUM
+#define CP110_TDM_INDEX			CP110_NUM
 #define CP110_SPI_BUS_ID(n)		((CP110_NUM * 2) + (n))
 
 / {
@@ -521,6 +522,7 @@
 			interrupts = <ICU_GRP_NSR 111 IRQ_TYPE_LEVEL_HIGH>,
 				     <ICU_GRP_NSR 112 IRQ_TYPE_LEVEL_HIGH>,
 				     <ICU_GRP_NSR 113 IRQ_TYPE_LEVEL_HIGH>;
+			cell-index = <CP110_TDM_INDEX>;
 			clock-names = "core", "axi";
 			clocks = <&CP110_LABEL(clk) 1 1>,
 				 <&CP110_LABEL(clk) 1 17>;
-- 
2.17.1

